libparsec/crates/client/src/certif/mod.rs

```diff
@@ -51,6 +51,82 @@ use libparsec_types::prelude::*;
 
 use crate::{event_bus::EventBus, ClientConfig};
 
+// The following values define an offset to be used when processing
+// a require greater timestamp error. They are different to give priority
+// to some kind of certificates (ex: a certificate that revokes a user
+// should have more chances to be accepted than a vlob update one)
+// The base offset is the average network round trip time to compensate
+// the first round trip.
+
+// An optimization could be to measure the real round trip time for each request.
+// The issue with that is that we tackling two distinct issues
+// 1. the client clock is lagging behind
+// 2. the client connection is degraded
+// The current approach is only compensating for case 1.
+// Case 2 would benefit from a offset depending on the real round trip time,
+// but if we base our estimate of the round trip time on the expected timestamp
+// provided by the server, case 2 would benefit from an unjustified priority, as
+// it would pass for a client with a very degraded connection (within the limits of the ballpark though).
+// That issue would not arise if we monitor the round trip time only from the client point of view.
+
+/// This value is used to increment the timestamp provided by the backend
+/// when a manifest restamping is required. This value should be kept small
+/// compared to the certificate stamp ahead value, so the certificate updates have
+/// priority over manifest updates.
```

Comment:
No because if the offset is small there is a greater chance that something else was acknowledged by the server.
For example, if a server answers with requireGreaterTimestamp(T0) to client A, then client B sends a new certificate at T0+x, and the server accepts it. When client A will answer if the offset is smaller or equal to x, the corresponding certificate will be refused, whereas if it's greater it will be accepted.

Reply:
I'll add that to the doc, it's a bit counter intuitive.